From 1311a55aae137970015be4251b0f2a1ce0f369f9 Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Fri, 26 Feb 2021 08:20:03 -0600
Subject: [PATCH 42/44] arm64: dts: renesas: beacon som: Reserve memory for
 lossy decompress

There is a check of memory about 48M in size that exists at address
0x54000000.  Its pre-allocated by TF-A, so Linux needs to reserve it
as well or there will be memory corruption.

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi b/arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi
index 81133b0f51da..e68e3b9eb3cb 100644
--- a/arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi
@@ -20,6 +20,12 @@
 		#size-cells = <2>;
 		ranges;
 
+		/* device specific region for Lossy Decompression */
+		lossy_decompress: linux,lossy_decompress@54000000 {
+			no-map;
+			reg = <0x00000000 0x54000000 0x0 0x03000000>;
+		};
+
 		/* global autoconfigured region for contiguous allocations */
 		linux,cma@58000000 {
 			compatible = "shared-dma-pool";
@@ -38,7 +44,7 @@
 
 	mmngr {
 		compatible = "renesas,mmngr";
-		memory-region = <&mmp_reserved>;
+		memory-region = <&mmp_reserved>, <&lossy_decompress>;
 	};
 
 	mmngrbuf {
-- 
2.17.1

